packages/backend-plugin-manager/src/scanner/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PackageRole } from '@backstage/cli-node';
+
+/**
+ * @public
+ */
+export interface ScannedPluginPackage {
+  location: URL;
+  manifest: ScannedPluginManifest;
+}
+
+/**
+ * @public
+ */
+export interface ScannedPluginManifest {
```

Comment:
Thoughts on using https://github.com/backstage/backstage/blob/73537f9cff8bb6fcdbbea3950b231a1a9e1e1d04/packages/cli-node/api-report.md#L16 instead of this?

Reply:
Good catch, Thanks.
I now use a type built from `BackstagePackageJson`, but with `main` and `backstage.role` made required.
This way it is assignable to `BackstagePackageJson`, but still at use-site we won't need to check for these fields to be defined, while this has already been checked during scanning.